pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -77,135 +45,63 @@ func (current *Resources) Remove(incoming Resources, all bool) {
-// another group of resources. If the resources are not able to fit,
-// an aggregated error will be returned with all exceeded resources.
-// If the current resources defines unlimited, then it will always fit.
-func (current *Resources) Fits(incoming Resources) error {
-	exceededResources := []string{}
-
-	// Define function for checking int resources to keep code DRY
-	checkResource := func(resource string, currentVal, incomingVal int) {
-		if currentVal != Unlimited && currentVal < incomingVal {
-			exceededResources = append(exceededResources, resource)
-		}
+func SubQuantity(c, i resource.Quantity) resource.Quantity {
+	// We don't expect this situation to happen by the same logic that is described in the Sub function.
+	if c.Equal(comparableUnlimitedQuantity) || i.Equal(comparableUnlimitedQuantity) {
+		return c
 	}
-
-	// Define function for checking quantity resources to keep code DRY
-	checkQuantityResource := func(resource string, currentVal, incomingVal resource.Quantity) {
-		if !currentVal.Equal(UnlimitedQuantity()) && currentVal.Cmp(incomingVal) < 0 {
-			exceededResources = append(exceededResources, resource)
-		}
+	c.Sub(i)
+	if c.CmpInt64(0) < 0 {
+		c.Set(0)
 	}
+	return c
+}
 
-	checkResource("Apps", current.Apps, incoming.Apps)
-	checkResource("Containers", current.Containers, incoming.Containers)
-	checkResource("Jobs", current.Jobs, incoming.Jobs)
-	checkResource("Volumes", current.Volumes, incoming.Volumes)
-	checkResource("Secrets", current.Secrets, incoming.Secrets)
-	checkResource("Images", current.Images, incoming.Images)
-	checkResource("Projects", current.Projects, incoming.Projects)
-
-	checkQuantityResource("VolumeStorage", current.VolumeStorage, incoming.VolumeStorage)
-	checkQuantityResource("Memory", current.Memory, incoming.Memory)
-	checkQuantityResource("Cpu", current.CPU, incoming.CPU)
-
-	// Build an aggregated error message for the exceeded resources
-	if len(exceededResources) > 0 {
-		return fmt.Errorf("%w: %s", ErrExceededResources, strings.Join(exceededResources, ", "))
+func Fits(current, incoming int) bool {
+	if current != Unlimited && current < incoming {
+		return false
 	}
+	return true
+}
 
-	return nil
+func FitsQuantity(current, incoming resource.Quantity) bool {
+	if !current.Equal(comparableUnlimitedQuantity) && current.Cmp(incoming) < 0 {
+		return false
+	}
+	return true
 }
 
-// NonEmptyString will return a string representation of the non-empty
-// Resources within the struct.
-func (current *Resources) NonEmptyString() string {
-	var resources []string
+// ResourceToString will return a string representation of the resource and value
+// if its value is greater than 0.
+func ResourcesToString(resources map[string]int, quantityResources map[string]resource.Quantity) string {
+	var resourceStrings []string
 
-	// Define function for checking int resources to keep code DRY
-	checkResource := func(resource string, value int) {
+	for resource, value := range resources {
```

Comment:
Similar comment to above: this will lead to inconsistent strings. Not sure if this will affect anything, but wanted to call it out just in case.

Reply:
@tylerslaton Curious what your thoughts are here.

Reply:
Ah, let me think on this. My bad, thought I covered this one already.

Reply:
Alright I think I came up with a good solution here. Now I'm using the `typed.Sorted` function here. This felt right as it keeps the simplicity of the function while maintaining determinism. Thoughts?

Reply:
That's fine. My overall opinion would be that we use the same approach in all of these cases (you uses an anonymous struct in the other case), but I won't hold up approval on that.